### DISCUSSION TITLE
Umstellung freetz-ng Wireguard nach Fritz!OS 7.50 Wireguard

### DISCUSSION THREAD

**Discussion by michael0815000** [Freetz-NG/freetz-ng#663] [Allgemein]
Hat schon jemand Erfahrungen gesammelt, der schon länger das von freetz-ng bereitgestellte Wireguard Paket im Einsatz hatte und dann auf eine Labor oder die finale FRITZ!OS Version 7.50 mit dem von AVM bereitgestellten Wireguard upgedatet hat?
Aktuell habe ich eine 7590 mit 154.07.29 rev92051 und ng-20417MA-9544d02e1 und noch das in freetz-ng enthaltene Wireguard-Paket im Einsatz

wireguard-tools 1.0.20210914
wireguard-linux-compat 1.0.20220627

Dieses Paket ist aber mit Version 7.50 nicht mehr verfügbar

--- wireguard-tools (not available, needs kernel modules)

Hat die Umstellung schon jemand ausprobiert und kann kurz seine Erfahrungen schildern?

**Comment by fda77** [comment]
Musst du source fragen. Hättest du schon während der 1 Jährigen labor-phase machen könne

**Reply by michael0815000** [reply]
Falls das hier irgendwie falsch rübergekommen sein sollte: Es sollte keinerlei Kritik darstellen, dass das Wireguard-Paket für die neue Version 7.50 noch nicht zur Verfügung steht. Ich war und bin sehr dankbar für Dein Engagement für freetz-ng.
Die Labor-Phase habe ich komplett ausgelassen, da die 7590 sehr rund lief und läuft und da beruflich und privat andere große Baustellen Vorrang hatten.
Ich habe die Sourcen jetzt mal bei AVM angefordert und gebe Bescheid, falls ich eine Rückmeldung erhalten sollte.

Nichtsdestotrotz gibt es vielleicht jemanden, der den Umstieg auf AVM Wireguard gemacht hat, und der ereit ist, seine Erfahrungen hier zu teilen.

**Reply by fda77** [reply]
Bei der Freetz-version kannst du alles genau so einstellen wie es dir passt. Ausserdem hast du es selbst compiliert und weisst was alles drin steckt, und was nicht. Es kann auch sein dass AVM manche Features mit der nächsten Version ohne Ankündigung beschneidet.
So ging es mit mit Voip, das ich seit FOS  7.2 nicht mehr mit Hybrid verwenden kann da ein Dhcp-Flag spontan fehlt und nicht reaktivierbar ist.
Ein guter Hersteller ist für mich einer die ohne viel Herumgehampel die Sourcen gleich mit der Firmware veröffentlicht...

**Comment by michael0815000** [comment]
Jetzt wollte ich gerade mitteilen, dass ich eine E-Mail erhalten habe, dass AVM den Quellcode für die 7590 FRITZ!OS 7.50 zur Verfügung gestellt hat, da sehe ich, dass Du das (natürlich) schon bemerkt und gleich eingearbeitet hast. Vielen Dank dafür!

Jetzt würde ich hoffen, dass sich an meiner funktionierenden Wireguard-Konfiguration durch das FRITZ!OS-Update nichts ändert und dass das von AVM bereitgestellte Wireguard dem nicht in die Quere kommt.
Ich werde berichten.

**Comment by fda77** [comment]
Falls du eh schon mit dem AVM wg herumexperiemtiert hast, vergleich doch mal die Geschwindigkeit. Es könnte sein dass (je Fritzbox modell) eine HW-Beschleunigung genutzt wird. kA ob AVM dazu was in den sourcen hat
Fall du aber nur einen gammelige Telekom Leitung mit 50 Mbit hast bringt das aber nichts, das sollte die 7590 auch so locker schaffen
